docs/design/services/dynamodb/high-level-library/prototype/option-1/sync/Prototype.java

```diff
@@ -0,0 +1,1026 @@
+     *                                                       .build())
+     *                                             .consistentRead(true)
+     *                                             .build());
+     *
+     *         // Times are stored as numbers, by default, so they can also be used as sort keys.
+     *         assert response.item().attribute("creation-time").isNumber();
+     *         assert response.item().attribute("creation-time").as(Instant.class).isBetween(Instant.now().minus(1, MINUTE),
+     *                                                                                       Instant.now());
+     *     } catch (NoSuchItemException e) {
+     *         System.out.println("Item was deleted between creation and retrieval.");
+     *         throw e;
+     *     }
+     * </code>
+     */
+    GetItemResponse getItem(GetItemRequest getRequest)
+            throws NoSuchItemException;
+
+    /**
+     * Invoke DynamoDB to retrieve an Item in this table.
+     *
+     * This will use the partition and sort keys from the provided object and convert the DynamoDB response to a Java object
+     * automatically using the default Object-to-Item {@link ItemAttributeValueConverter}, unless an alternate converter
+     * has been overridden for the provided type.
+     *
+     * This method is optimized for performance, and provides no additional response data. For additional options
+     * like consistent reads, see {@link #getObject(GetObjectRequest)}.
+     *
+     * Usage Example:
+     * <code>
+     *     public class MyItem {
+     *         @Attribute("partition-key")
+     *         @Index(AttributeIndexType.PARTITION_KEY)
+     *         private String partitionKey;
+     *
+     *         @Attribute("creation-time")
+     *         private Instant creationTime;
+     *
+     *         public String getPartitionKey() { return this.partitionKey; }
+     *         public Instant getCreationTime() { return this.creationTime; }
+     *         public void setPartitionKey(String partitionKey) { this.partitionKey = partitionKey; }
+     *         public void setCreationTime(Instant creationTime) { this.creationTime = creationTime; }
+     *     }
+     *
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         Table table = client.getTable("my-table");
+     *
+     *         UUID id = UUID.randomUUID();
+     *
+     *         MyItem itemToCreate = new MyItem();
+     *         itemToCreate.setPartitionKey(id);
+     *         itemToCreate.setCreationTime(Instant.now());
+     *
+     *         table.putObject(itemToCreate);
+     *
+     *         // Wait a little bit, because getObject is eventually consistent by default.
+     *         Thread.sleep(5_000);
+     *
+     *         MyItem itemToRetrieve = new MyItem();
+     *         itemToRetrieve.setPartitionKey(id);
+     *
+     *         MyItem retrievedItem = table.getObject(itemToRetrieve);
```

Comment:
Why take in the whole object instead of just the keys?

Reply:
I had that originally. What syntax would you propose?

Reply:
I don't really have a specific one in mind. I just thought might be a little confusing with passing in the item type since it's not super clear that we just want the keys. I don't feel that strongly about it though, and do like that this is concise.

Reply:
It's tricky because retrieval needs the attribute value names for the keys, and I wanted to avoid needing to use the POJO APIs for writing and the Item APIs for retrieving.

I agree that it's not super clear, but this felt the simplest way of fixing the above problem. If there's another way, I'm open to it!